src/openrct2/peep/Guest.cpp

```diff
@@ -694,6 +694,181 @@ int32_t Guest::CheckEasterEggName(int32_t index) const
     return _stricmp(buffer, gPeepEasterEggNames[index]) == 0;
 }
 
+void Guest::UpdateStatistics()
+{
+    // Idle peep happiness tends towards 127 (50%).
+    if (HappinessTarget >= 128)
+        HappinessTarget--;
+    else
+        HappinessTarget++;
+
+    NauseaTarget = std::max(NauseaTarget - 2, 0);
+
+    if (Energy <= 50)
+    {
+        Energy = std::max(Energy - 2, 0);
+    }
+
+    if (Hunger < 10)
+    {
+        Hunger = std::max(Hunger - 1, 0);
+    }
+
+    if (Thirst < 10)
+    {
+        Thirst = std::max(Thirst - 1, 0);
+    }
+
+    if (Toilet >= 195)
+    {
+        Toilet--;
+    }
+
+    if (State == PEEP_STATE_WALKING && NauseaTarget >= 128)
+    {
+        if ((scenario_rand() & 0xFF) <= static_cast<uint8_t>((Nausea - 128) / 2))
+        {
+            if (Action >= PEEP_ACTION_NONE_1)
+            {
+                Action = PEEP_ACTION_THROW_UP;
+                ActionFrame = 0;
+                ActionSpriteImageOffset = 0;
+                UpdateCurrentActionSpriteType();
+            }
+        }
+    }
+
+    // Remaining content is executed every call.
```

Comment:
You have changed the logic here. This needs to be split up into two functions as this second part from here is called every time but the first part is only called sometimes. That's what this comment is for.

Reply:
You are right, I've pretty much reverted all my changes. I haven't found a proper name for these functions.